### PR TITLE
4.1.0

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by jordisala1991** [comment]
Phpstan and psalm failing

**Comment by VincentLanglet** [comment]
> Phpstan and psalm failing

Fixed

File: src/FieldDescription/FieldDescriptionFactory.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function create(string $class, string $name, array $options = []): FieldD
      * @phpstan-param class-string $baseClass
      *
      * @phpstan-return array{
-     *      ClassMetadata,
+     *      ClassMetadata<object>,
```
Comment (jordisala1991) [inline src/FieldDescription/FieldDescriptionFactory.php:52]: isnt this related to the class string above?

Edit: now I read the full method and it isnt really related

**Review by jordisala1991 — APPROVED** [review]

File: src/FieldDescription/FieldDescriptionFactory.php (hunk 1 above)
Comment (VincentLanglet) [reply src/FieldDescription/FieldDescriptionFactory.php:52]: Not always, with `foo.bar.baz`, you give Foo, you end with ClassMetadata of Baz.

**Pull request merged**